src/Streamly/Internal/Data/Parser.hs

```diff
@@ -570,6 +570,11 @@ groupBy = undefined
 
 -- | Match the given sequence of elements using the given comparison function.
 --
+-- >>> S.parse $ S.eqBy (==) "string" $ S.fromList "string"
+--
+-- >>> S.parse $ S.eqBy (==) "mismatch" $ S.fromList "match"
+-- > *** Exception: ParseError "eqBy: failed, yet to match 6 elements"
```

Comment:
There is something wrong with the count? yet to match 6 elements?

Reply:
The word "match" isn't matched yet, and it has 6 letters in it.